### PR TITLE
You can now push Wraith Spectacles up

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan, Dagdammit
rscadd: You can now push Wraith Spectacles up to avoid vision damage, but lose xray vision.
wip: Do note that flicking them on and off very quickly may cause you to lose vision rather quickly.
/:cl:

Oh hey, this is really useful!
Credit to @Dagdammit for the idea

Status effects are now a per-mob list instead of a global list, and are properly lazy-d.

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -685,10 +685,21 @@ BLIND     // can't see anything
...
 
-/obj/item/clothing/proc/weldingvisortoggle()			//Malk: proc to toggle welding visors on helmets, masks, goggles, etc.
+/obj/item/clothing/proc/weldingvisortoggle() //proc to toggle welding visors on helmets, masks, goggles, etc.
 	if(!can_use(usr))
-		return
+		return FALSE
+
+	visor_toggling()
 
+	if(istype(usr, /mob/living/carbon))
```
Comment (Iamgoofball) [inline code/modules/clothing/clothing.dm]: >usr

fix it

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -685,10 +685,21 @@ BLIND     // can't see anything
...
+		var/mob/living/carbon/C = usr
+		C.head_update(src, forced = 1)
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.UpdateButtonIcon()
+	return TRUE
+
+/obj/item/clothing/proc/visor_toggling() //handles all the actual toggling of flags
 	up ^= 1
```
Comment (Iamgoofball) [inline code/modules/clothing/clothing.dm]: what the fuck is this + carbon dioxide

File: code/modules/clothing/glasses/glasses.dm [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
 
+/obj/item/clothing/glasses/visor_toggling()
+	..()
+	vision_flags ^= initial(vision_flags)
+	darkness_view ^= initial(darkness_view)
+	invis_view ^= initial(invis_view)
+
+/obj/item/clothing/glasses/weldingvisortoggle()
+	. = ..()
+	if(. && usr)
```
Comment (Iamgoofball) [inline code/modules/clothing/glasses/glasses.dm]: >usr

fix it

**Comment by jughu** [comment]
Powercreep pr

**Comment by Dagdammit** [comment]
Could counter with like 5 eye damage every time xray effect starts, to
discourage constantly flicking it on for a microsecond.

**Comment by AdamElTablawy** [comment]
honestly i don't see the point of the eye damage thing

it's already an item you can't wear in public without getting shot and spaced and you can counteract the effects by chugging fucktons of oculine

not to mention other antagonists (lings, for example) will get thermals, for free that can be kept on however long they want

if you want this to be used, honestly, just remove the eye damage thing since it's only good for seeing things and literally any other antag can do that

blood cult can astral communion, almost every antag has thermals of some kind, true sight isn't too good for clock cult considering component cost in early game and late game it won't even matter + unlike most antags you can't wear this shit in public without getting your ass canned

**Comment by ChangelingRain** [comment]
Wraith specs are literally free, and, besides the eye damage thing, have no downside, and thermals, ling included, fuck you up if you get flashed.
Plus, no, you can't actually counteract the effects by drinking oculine; I prevented that a while ago.

**Pull request merged**